solutions/ibm-catalog/sap-s4hana-bw4hana/main.tf

```diff
@@ -0,0 +1,306 @@
+    cos_resource_instance_id = local.cos_resource_instance_id
+    cos_bucket_name          = var.ibmcloud_cos_configuration.cos_bucket_name
+    cos_dir_name             = var.ibmcloud_cos_configuration.cos_hana_software_path
+    download_dir_path        = local.powervs_network_services_config.nfs.nfs_client_path
+  }
+
+  ibmcloud_cos_solution_configuration = {
+    cos_apikey               = local.cos_apikey
+    cos_region               = var.ibmcloud_cos_configuration.cos_region
+    cos_resource_instance_id = local.cos_resource_instance_id
+    cos_bucket_name          = var.ibmcloud_cos_configuration.cos_bucket_name
+    cos_dir_name             = var.ibmcloud_cos_configuration.cos_solution_software_path
+    download_dir_path        = local.powervs_network_services_config.nfs.nfs_client_path
+  }
+
+  ibmcloud_cos_monitoring_configuration = {
+    cos_apikey               = local.cos_apikey
+    cos_region               = var.ibmcloud_cos_configuration.cos_region
+    cos_resource_instance_id = local.cos_resource_instance_id
+    cos_bucket_name          = var.ibmcloud_cos_configuration.cos_bucket_name
+    cos_dir_name             = var.ibmcloud_cos_configuration.cos_monitoring_software_path
+    download_dir_path        = local.powervs_network_services_config.nfs.nfs_client_path
+  }
+}
+
+module "ibmcloud_cos_download_hana_binaries" {
+  source                     = "../../../modules/ibmcloud-cos"
+  count                      = module.standard.network_services_config.nfs.enable ? 1 : 0
+  access_host_or_ip          = module.standard.access_host_or_ip
+  target_server_ip           = module.standard.ansible_host_or_ip
+  ssh_private_key            = var.ssh_private_key
+  ibmcloud_cos_configuration = local.ibmcloud_cos_hana_configuration
+}
+
+module "ibmcloud_cos_download_netweaver_binaries" {
+  source = "../../../modules/ibmcloud-cos"
+
+  depends_on = [module.ibmcloud_cos_download_hana_binaries]
+
+  access_host_or_ip          = module.standard.access_host_or_ip
+  target_server_ip           = module.standard.ansible_host_or_ip
+  ssh_private_key            = var.ssh_private_key
+  ibmcloud_cos_configuration = local.ibmcloud_cos_solution_configuration
+}
+
+locals {
+  monitoring_instance = module.standard.monitoring_instance
+}
+
+module "ibmcloud_cos_download_monitoring_binaries" {
+  source     = "../../../modules/ibmcloud-cos"
+  depends_on = [module.ibmcloud_cos_download_netweaver_binaries]
+  count      = local.monitoring_instance.enable ? 1 : 0
+
+  access_host_or_ip          = module.standard.access_host_or_ip
+  target_server_ip           = module.standard.ansible_host_or_ip
+  ssh_private_key            = var.ssh_private_key
+  ibmcloud_cos_configuration = local.ibmcloud_cos_monitoring_configuration
+}
+
+#####################################################
+# Ansible vars validation
+#####################################################
+
+locals {
```

Comment:
this validation can happen in variables.tf . back then we didn't have cross variable validation and hence had to use this dirty hack. Please move it to variables.tf file under sap_hana_vars